crypto/crypto/src/multi_ed25519.rs

```diff
@@ -141,15 +137,15 @@ impl SigningKey for MultiEd25519PrivateKey {
     // Sign a message with the minimum amount of keys to meet threshold (starting from left-most keys).
     fn sign_message(&self, message: &HashValue) -> MultiEd25519Signature {
         let mut signatures: Vec<Ed25519Signature> = Vec::with_capacity(self.threshold as usize);
-        let mut bit_vector = BitVec::from_elem(MAX_NUM_OF_KEYS, false);
+        let mut bit_vector = 0u32;
 
         signatures.extend(
             self.private_keys
                 .iter()
                 .take(self.threshold as usize)
                 .enumerate()
                 .map(|(i, item)| {
-                    bit_vector.set(i, true);
+                    bit_vector |= SINGLE_SET_BIT_NUMBERS[i];
```

Comment:
Would `bit_vector |= (0 << i)` work here if you willing to interpret the bits from right to left instead (or `bit_vector |= (16777216 >> i)` if you want to stick with left to right)? That seems simpler than the table, and probably more efficient given that Rust can't safely inline the value of `SINGLE_SET_BIT_NUMBERS[i]`... Similar for `get_bit_at`.

Reply:
This will not guarantee the left to right bit indexing, i.e., to make it compatible with a simple `[u8; 4]`. The only reason I picked `u32` was for convenience (i.e., it provides an api for `count_ones()` already and due to more succinct serialization (no lengths required). 

However, I'm open in discussing the use of `[u8; 4]` vs `u32`, because serializing in little endian, but then reading bits from left to right might cause confusion when communicating it.